lib/src/component_declaration/redux_component.dart

```diff
@@ -0,0 +1,159 @@
+        B extends Builder<V, B>,
+        A extends ReduxActions,
+        T extends ReduxUiProps<V, B, A>,
+        Substate> extends UiComponent<T> with _ReduxComponentMixin<V, B, A, T, Substate> {
+  @mustCallSuper
+  @override
+  void componentWillMount() {
+    super.componentWillMount();
+    _setUpSub();
+  }
+
+  @mustCallSuper
+  @override
+  void componentWillReceiveProps(Map nextProps) {
+    super.componentWillReceiveProps(nextProps);
+    _tearDownSub();
+  }
+
+  @mustCallSuper
+  @override
+  void componentWillUpdate(Map nextProps, Map nextState) {
+    if (_storeSub == null) _setUpSub();
+  }
+
+  @mustCallSuper
+  @override
+  void componentWillUnmount() {
+    super.componentWillUnmount();
+    _tearDownSub();
+  }
+}
+
+abstract class _ReduxComponentMixin<
+    V extends Built<V, B>,
+    B extends Builder<V, B>,
+    A extends ReduxActions,
+    T extends ReduxUiProps<V, B, A>,
+    Substate> implements UiComponent<T> {
+  Substate _connectedState;
+
+  /// The substate values of the redux store that this component component subscribes to.
+  ///
+  /// It is reccommened to use this instead of `props.store.state`,
+  /// if you need to access other state values from the store then they should be connected with [connect].
+  ///
+  /// Related: [connect]
+  Substate get connectedState => _connectedState;
+
+  /// Subscribe to changes to the values from the redux store that this component cares about.
+  ///
+  /// By default components will redraw based on any value change in the redux store.
+  ///
+  /// Example where there are three state values but your component should only re-render when two of them update:
+  ///
+  ///     abstract class MyState implements Built<MyState, MyStateBuilder> {
+  ///       factory MyState({
+  ///         int foo = 0,
+  ///         int bar = 0,
+  ///         int baz = 0,
+  ///       }) => _$MyState._(
+  ///         foo: foo,
+  ///         bar: bar,
+  ///         baz: baz,
+  ///       );
+  ///       MyState._();
+  ///
+  ///       int get foo;
+  ///       int get bar;
+  ///       int get baz;
+  ///     }
+  ///
+  ///     abstract class DataComponentCaresAbout implements Built<DataComponentCaresAbout, DataComponentCaresAboutBuilder> {
+  ///       factory DataComponentCaresAbout({int foo, int bar}) => _$DataComponentCaresAbout._(foo: foo, bar: bar);
+  ///       DataComponentCaresAbout._();
+  ///
+  ///       int get foo;
+  ///       int get bar;
+  ///     }
+  ///
+  ///     MyComponent extends ReduxUiComponent<MyState, MyStateBuilder, MyActions, MyProps, DataComponentCaresAbout> {
+  ///       @override
+  ///       DataComponentCaresAbout connect(state) => new DataComponentCaresAbout(foo: state.foo, bar: state.bar);
+  ///
+  ///       render() {
+  ///         return Dom.dov()(
+  ///           connectedState.foo,
+  ///           connectedState.bar,
+  ///         );
+  ///       }
+  ///     }
+  ///
+  /// Related: [connectedState]
+  Substate connect(V state);
```

Comment:
I think it would be worth adding to the docstring that Substate must be a comparable object to avoid unnecessary redraws. Add a recommendation that Substate should be either primitive, built_value, built_collection, or other type of object that overrides the == operator.

Reply:
as well as a similar blurb to the ReduxUiComponent docstring